packages/signature-controller/src/SignatureController.ts

```diff
@@ -382,12 +395,48 @@ export class SignatureController extends BaseController<
     const chainId = this.#getChainId(request);
     const internalAccounts = this.#getInternalAccounts();
 
+    const isRequestDelegation = isDelegationRequest({
+      data: messageParams.data,
+    });
+
+    let decodedPermission: DecodedPermission | undefined;
+    // Only decode the permission for SignTypedDataVersion.V4, as all other versions will be rejected.
+    if (
+      isRequestDelegation &&
+      (version as SignTypedDataVersion) === SignTypedDataVersion.V4
+    ) {
+      try {
+        if (request.origin) {
+          // no point in trying to decode the permission if the origin is not set
+          decodedPermission = await decodePermissionFromRequest({
+            origin: request.origin,
+            messageParams,
+            messenger: this.messagingSystem,
+          });
+        }
+      } catch (error) {
+        // we ignore this error, because it simply means the request could not
+        // be decoded into a permission in which case we will throw below
+        log((error as Error).message);
+      }
+      if (!decodedPermission) {
```

Comment:
Isn't this duplicating the check in the subsequent validation?

What if we want to decode permissions for internal delegations in future?